documentation/book/proc-creating-kafka-user-scram-sha.adoc

```diff
@@ -0,0 +1,70 @@
+// Module included in the following assemblies:
+//
+// assembly-using-the-user-operator.adoc
+
+[id='proc-creating-kafka-user-scram-{context}']
+= Creating a Kafka user with SCRAM SHA authentication
+
+.Prerequisites
+
+* A running Kafka cluster configured with a listener using SCRAM SHA authentication.
+* A running User Operator.
+
+.Procedure
+
+. Prepare a YAML file containing the `KafkaUser` to be created.
++
+.An example `KafkaUser`
+[source,yaml,subs="attributes+"]
+----
+apiVersion: {KafkaUserApiVersion}
+kind: KafkaUser
+metadata:
+  name: my-user
+  labels:
+    strimzi.io/cluster: my-cluster
+spec:
+  authentication:
+    type: scram-sha-512
+  authorization:
+    type: simple
+    acls:
+      - resource:
+          type: topic
+          name: my-topic
+          patternType: literal
+        operation: Read
+      - resource:
+          type: topic
+          name: my-topic
+          patternType: literal
+        operation: Describe
+      - resource:
+          type: group
+          name: my-group
+          patternType: literal
+        operation: Read
+----
+
+. Create the `KafkaUser` resource in {ProductPlatformName}.
++
+ifdef::Kubernetes[]
+On {KubernetesName} this can be done using `kubectl apply`:
++
+[source,shell,subs=+quotes]
+kubectl apply -f _<your-file>_
++
+endif::Kubernetes[]
+On {OpenShiftName} this can be done using `oc apply`:
++
+[source,shell,subs=+quotes]
+oc apply -f _<your-file>_
+
+. Use the credentials from the secret `my-user` in your application
```

Comment:
Should we somewhere add some information abotu the passwords we generate? Length, characters etc.?

Reply:
Is a procedure the right place to document that?

The current password length is an arbitrary choice, but right right it probably is worthy of documenting. 

The characters are currently `[a-zA-z0-9]`, Unicode should work, but since since Kafka doesn't apply SASLPrep, it could cause problems when trying to use the same password, but with a Kafka cluster configured for SCRAM delegation. So I'm inclined to say we should support only ASCII.

Should this stuff be configurable?

Reply:
I'm fine to keep this password as ascii only ... users can place requests when they want more. I think at this point we should just document it somewhere. I agree the procedure is not the right place - I'm not sure why I raised it exactly here.